src/a_star.h

```diff
@@ -0,0 +1,58 @@
+#ifndef A_STAR_H
+#define A_STAR_H
+
+#include "ros/ros.h"
+#include "nav_msgs/OccupancyGrid.h"
+#include "nav_msgs/Odometry.h"
+#include <vector>
+#include <queue>
+#include <unordered_set>
+
+class Node{
+    public:
+    Node(int x, int y, Node* parent);
+    private:
+    geometry_msgs::Point position;
+    Node* parent;

```

Comment:
We don't ever want to modify parents through their children. `const Node* parent`.